content/chapters/io/lecture/slides/linux-files.md

```diff
@@ -0,0 +1,110 @@
+## Linux Files
+
+---
+
+### I/O Devices
+
+- A hardware component that communicates with our application through bytes
+- Various devices fit this description:
+  - screen
+  - keyboard
+  - mouse
+  - hard disk
+  - sensors
+
+How does the OS operates them?
+
+----
+
+### Working with I/O Devices
+
+- Linux abstracts an I/O device as a **special file**
+- There are two categories of device files:
+  - **char device file** are slower and manage small amount of data
+    - e.g.: mouse, keyboard
+  - **block device file** manage large data volume organized in blocks
+    - e.g.: hard disk
+- Device special files are in `/dev`
+
+----
+
+### Device Special Files
+
+- The letter before permissions describes the file type
+
+```console
+student@OS:~$ ls -l /dev/
+crw-rw-rw-   1 root root      1,   3 nov  6 20:31 null
+crw-rw-rw-   1 root root      1,   8 nov  6 20:31 random
+brw-rw----   1 root disk      8,   1 nov  6 20:31 sda1
+brw-rw----   1 root disk      8,   2 nov  6 20:31 sda2
+brw-rw----   1 root disk      8,   5 nov  6 20:31 sda5
+brw-rw----   1 root disk      8,   6 nov  6 20:31 sda6
+crw-rw-rw-   1 root tty       5,   0 nov  6 20:31 tty
+crw-rw-rw-   1 root root      1,   9 nov  6 20:31 urandom
+crw-rw----+  1 root video    81,   0 nov  6 20:31 video0
+crw-rw----+  1 root video    81,   1 nov  6 20:31 video1
+crw-rw-rw-   1 root root      1,   5 nov  6 20:31 zero
+...
+```
+
+----
+
+### Persistent Data
+
+- Going a step further, we use files to represent and organize persistent data
+  - **regular files** store data
+  - **directories** construct a file hierarchy
+  - **symbolic links** referrence other files (shortcuts)
+- File hierarchies dramatically reduce search times
+
+----
+
+### Interprocess Communication
```

Comment:
This has to be a separate section. Mentioned the two types of endpoints. Those endpoints backed by storage / devices (file interface) and those endpoints pointing to another process (socket interface).